### PR TITLE
Split MengeCore.dll/a into two libraries

### LINKED ISSUE #34: mengeCore.dll is too large
The menge core currently encompasses simulation, visualization, command-line parsing, etc.  This means if components (e.g., simulation) would be applied elsewhere, we have to drag along all of the dynamic libraries that mengeCore.dll is currently dependent on. 

It would be better if menge was decomposed more strategically so that only those components of interest would be used.

### PR DESCRIPTION
**Pull request by curds01**
This PR does several things:

1) Creates the MengeVis and MengeCore libraries. The MengeCore library has no external dependencies.  The MengeVis library has various external dependencies to handle images, fonts, etc.,  The core library can be embedded in other systems without dragging along a lot of additional (and possibly unnecessary) libraries.
2) Add an initial c-style API to facilitate integration.
3) Remove VS 2005 project.
4) Advance version to 0.9.0
5) Scrub coding style into a more consistent style.

Fixes #34

### CODE REVIEW

**Comment by chraibi** [comment]
How can I pull this merge request?

**Comment by curds01** [comment]
If you look to the right of the "Merge pull request" button, there is a link for `command line instructions`.  It will give you precise instructions for creating a local branch and checking out from my branch into that local branch.

**Comment by chraibi** [comment]
After having to fix two small "errors" 

In this [line](https://github.com/curds01/Menge/blob/f69b1451e4a5b53f3ed37de04521d51edadca02e/src/Menge/MengeCore/resources/PathPlanner.cpp#L83) 

`endt` --> `end`

and two places in `CorePluginEngine.cpp` where `>>` should be replaced by `> >` (space inbetween)

```
-BasePluginEngine<CorePluginEngine, Plugin<CorePluginEngine>>(), _simDB( simDB ) {                    
+BasePluginEngine<CorePluginEngine, Plugin<CorePluginEngine> >(), _simDB( simDB ) {                   

```
class MENGE_API CorePluginEngine : public BasePluginEngine<CorePluginEngine,                                 
-  Plugin<CorePluginEngine>> {   
+  Plugin<CorePluginEngine> > {
```

I get now link errors 

<img width="1280" alt="screen shot 2017-02-12 at 16 34 21" src="https://cloud.githubusercontent.com/assets/5772973/22863515/c6a3b6fa-f141-11e6-8835-8b5a1dd0b2cd.png">

I can investigate this next days. Now we have Sunday here. 
(and a new president!)

**Comment by curds01** [comment]
I just submitted a new commit that addresses several issues.  I have high hopes it might alleviate your linker problem.  Although, to be perfectly frank, your linker issue is a *bit* mysterious.  I compiled against both g++ and clang++.  I had what appeared to be a *related* linker problem.  I addressed that.  So, fingers crossed.

(Wanna trade presidents? :D)

**Comment by chraibi** [comment]
Good news! The error is solved. 
Here are other new errors: 

```
modified   src/Menge/MengeVis/PluginEngine/VisPluginEngine.cpp                                                                 
@@ -24,7 +24,7 @@ namespace MengeVis {                                                                                         
                /////////////////////////////////////////////////////////////////////                                          
                                                                                                                               
                VisPluginEngine::VisPluginEngine() :                                                                           
-                       BasePluginEngine<VisPluginEngine, Plugin<VisPluginEngine>>() {                                         
+                       BasePluginEngine<VisPluginEngine, Plugin<VisPluginEngine> >() {                                        
```
modified   src/Menge/MengeVis/PluginEngine/VisPluginEngine.h                                                                   
@@ -28,7 +28,7 @@ namespace MengeVis {                                                                                         
                 */                                                                                                            
                class MENGEVIS_API VisPluginEngine :                                                                           
                        public Menge::PluginEngine::BasePluginEngine<VisPluginEngine,                                          
-                                                                      Menge::PluginEngine::Plugin<VisPluginEngine>> {         
+                                                                      Menge::PluginEngine::Plugin<VisPluginEngine> > {        
           
```

Sorry I don't know how to push to a merge request, so I ll just paste here the diffs. 

Now we have a new Linker error. 

<img width="1280" alt="screen shot 2017-02-12 at 21 32 36" src="https://cloud.githubusercontent.com/assets/5772973/22865847/2b41ce38-f16c-11e6-8a9c-494b69d27ceb.png">

See also this log for the complete error. 
[log_Menge.txt](https://github.com/MengeCrowdSim/Menge/files/769612/log_Menge.txt)

This `MengeVis` is something new, right? 


```

**Comment by curds01** [comment]
So, I've addressed the diff.

The linker errors are killing me.  I'd hoped that by using clang on Ubuntu, I could reasonably approximate what you're experiencing.  But I don't get *any* of these artifacts.  Can you remind me what version of clang you're using?

**Comment by chraibi** [comment]
screen shot 2017-02-12 at 23 28 39

**Comment by curds01** [comment]
Apple is killing me.

Now I have to figure out how to compare arbitrarily different version codes: Apple LLVM 8.0.0 vs Ubuntu's 3.8.0.  Siiiiiigh.

**Comment by curds01** [comment]
I think I've solved the linker issue.  (At least, I hope.)

File: projects/g++/Menge/MengeVis/CMakeLists.txt [hunk 1]
```diff
@@ -56,6 +56,6 @@ add_library(
 	${source_files}
 )
 
-target_link_libraries ( mengeVis  dl tinyxml ${OPENGL_LIBRARIES} ${SDL_LIBRARY} ${PNG_LIBRARY}  ${SDLIMAGE_LIBRARY}  ${SDLTTF_LIBRARY} )
+target_link_libraries ( mengeVis mengeCore dl tinyxml ${OPENGL_LIBRARIES} ${SDL_LIBRARY} ${PNG_LIBRARY}  ${SDLIMAGE_LIBRARY}  ${SDLTTF_LIBRARY} )
```
Comment (chraibi) [inline projects/g++/Menge/MengeVis/CMakeLists.txt]: This seems to be mandatory. I don't understand how you could get this compiled under under windows

**Comment by chraibi** [comment]
Ok Sean. You are doing well. Problem solved. 

Now we are at 87% 

screen shot 2017-02-13 at 09 34 21

**Comment by curds01** [comment]
The most obvious guess, based on this message, is that the linking is missing a reference to `mengeVis`.  I've added the link dependency (to this as well as `AgtZanlungo`.)  The `terrain` plug-in already had it and I don't think any of the other plug ins require it.

It's insane how inconsistent build systems are.  Clearly, on your mac, the build system is being *very* picky about build-time consistencies; it wants to know that the hanging symbols can actually be dereferenced into libraries.  So, by omitting the library dependency, it can't confirm that it's a valid binary.  The other systems appear to have decided that it's a run-time problem.  Check the libraries if available, but don't worry about it.  And that's g++ *and* clang++ on ubuntu.

**Comment by chraibi** [comment]
Just to say that compiling using a fresh Ubuntu Image works fine! 
So, as you said the different systems the building process is different.

**Comment by curds01** [comment]
So, does this version build on your mac.  Are we good to go?

**Comment by chraibi** [comment]
Negative. The above mentioned linker error persists. 
Sorry I did not manage to check why. Maybe in the week end.

**Comment by curds01** [comment]
It persists since my comment indicating that I thought I fixed it? You'll have to pull again.

**Comment by chraibi** [comment]
```
HEAD is now at 70e182c... Another round of mac osx build fixes
```

and we are at 87%. 

```
[ 87%] Linking CXX shared library /Users/chraibi/Workspace/Menge/Exe/plugins/libhelbingPed.dylib
Undefined symbols for architecture x86_64:
  "MengeVis::SceneGraph::SelectContext::selectGL(MengeVis::SceneGraph::GLScene const*, MengeVis::SceneGraph::GLCamera const&,
int, int, int*)", referenced from:
```

**Comment by chraibi** [comment]
Ok my dear here is the solution 
screen shot 2017-02-16 at 13 30 41

**Comment by curds01** [comment]
Stupidly, that's *exactly* the fix I'd done several days ago.  I thought I'd commited and pushed it, but apparently not.  Sorry for wasting your time.

**Comment by chraibi** [comment]
👍

**Pull request merged**